src/components/Input/FileInput.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { ChangeEvent } from 'react';
+import Input from './Input';
+
+type FileInputProps = {
+  onChange: (files: FileList | null) => void;
+  type: 'file';
+};
+
+class FileInput extends React.Component<FileInputProps> {
+  handleChange = (changeEvent: ChangeEvent) => {
+    const files = (changeEvent.target as HTMLInputElement).files;
+    if (this.props.onChange) {
+      this.props.onChange(files);
+    }
```

Comment:
Fyi, we could use the optional operator here:
```suggestion
    this.props.onChange?.(files);
```
```suggestion
    if (this.props.onChange) {
      this.props.onChange(files);
    }
```